src/BrewingRecipe.cpp

```diff
@@ -0,0 +1,324 @@
+
+#include "Globals.h"  // NOTE: MSVC stupidness requires this to be the same across all modules
+
+#include "BrewingRecipe.h"
+#include "Item.h"
+
+#include <fstream>
+
+#define BREWING_RECIPE_FILE "brewing.txt"
+
+
+
+
+
+typedef std::list<cBrewingRecipe::cRecipe> RecipeList;
+
+
+
+
+
+struct cBrewingRecipe::sBrewingRecipeState
+{
+	RecipeList Recipes;
+};
+
+
+
+
+
+cBrewingRecipe::cBrewingRecipe()
+	: m_pState(new sBrewingRecipeState)
+{
+	ReloadRecipes();
+}
+
+
+
+
+
+cBrewingRecipe::~cBrewingRecipe()
+{
+	ClearRecipes();
+	delete m_pState;
+	m_pState = nullptr;
+}
+
+
+
+
+
+void cBrewingRecipe::ReloadRecipes(void)
+{
+	ClearRecipes();
+	LOGD("Loading brewing recipes...");
+
+	std::ifstream f(BREWING_RECIPE_FILE, std::ios::in);
+	if (!f.good())
+	{
+		LOG("Could not open the brewing recipes file \"%s\". No brewing recipes are available.", BREWING_RECIPE_FILE);
+		return;
+	}
+
+	unsigned int LineNum = 0;
+	AString ParsingLine;
+
+	while (std::getline(f, ParsingLine))
+	{
+		LineNum++;
+		if (ParsingLine.empty())
+		{
+			continue;
+		}
+
+		// Remove comments from the line:
+		size_t FirstCommentSymbol = ParsingLine.find('#');
+		if ((FirstCommentSymbol != AString::npos) && (FirstCommentSymbol != 0))
+		{
+			ParsingLine.erase(ParsingLine.begin() + static_cast<const long>(FirstCommentSymbol), ParsingLine.end());
```

Comment:
My mistake, std::advance increments an lvalue. But its probably still worth using even with the extra two statements to first declare and the increment the iterator, given that the correct form of that cast is `static_cast<std::iterator_traits<AString::iterator>::difference_type(FirstCommentSymbol)`